src/animal/animal.providers.ts

```diff
@@ -0,0 +1,2 @@
+export const ANIMAL_SERVICE_KEY = 'ANIMAL_SERVICE';
+export const ANIMAL_REPOSITORY_KEY = 'ANIMAL_REPOSITORY';
```

Comment:
Don't place all tokens in one provider file. Use a file for each kind of Injectable, the nearest as possible to the declaration. Furthermore, use the `Symbol` and the reference to the interface:
```
export const ANIMAL_SERVICE_KEY = Symbol.for('IAnimalService')
```